src/modules/feedback-reports/index.ts

```diff
@@ -0,0 +1,264 @@
+        'FeedbackReports',
+        (FeedbackReports) =>
+            FeedbackReports.updateOne(
+                {
+                    _id: new ObjectID(reportId),
+                    'meta.createdBy._id': new ObjectID(user._id),
+                },
+                {
+                    $set: {
+                        'meta.updatedAt': new Date(),
+                        'meta.updatedBy': user,
+                        description: newDescription,
+                    },
+                },
+                {
+                    upsert: false,
+                }
+            )
+    );
+
+    if (modifiedCount === 0)
+        throw createHttpError(
+            401,
+            'You must be the creator in order to modify the feedback report'
+        );
+}
+
+// TODO: extend this to write to a trash collection rather than actually delete
+/**
+ * @description Deletes a feedback report by Id
+ * @param {string} reportId - Id of the feedback report to delete
+ * @returns {Promise} Void promise
+ * @throws {Error} If provided report id is invalid or if specified feedback report to delete does not exist
+ */
+export async function deleteFeedbackReport(reportId: string) {
+    if (!ObjectID.isValid(reportId))
+        throw createHttpError(400, 'Invalid report id provided');
+
+    const { deletedCount } = await useCollection(
+        'FeedbackReports',
+        (FeedbackReports) =>
+            FeedbackReports.deleteOne({ _id: new ObjectID(reportId) })
+    );
+
+    if (deletedCount === 0)
+        throw createHttpError(404, 'Feedback report not found');
+}
+
+/**
+ * @description Sets the resolved attribute of a feedback report to the new resolved status provided
+ * @param {string} reportId - Id of the report
+ * @param {boolean} newResolvedStatus - New resolved status
+ * @param {User} user - User object of updater
+ * @returns {Promise} Void promise
+ * @throws {Error} If provided report id is invalid or if unable to update the status of the specified feedback report
+ */
+export async function updateResolvedStatus(
+    reportId: string,
+    newResolvedStatus: boolean,
+    user: User
+) {
+    if (!ObjectID.isValid(reportId))
+        throw createHttpError(400, 'Invalid report id provided');
+
+    const { modifiedCount } = await useCollection(
+        'FeedbackReports',
+        (FeedbackReports) =>
+            FeedbackReports.updateOne(
+                {
+                    _id: new ObjectID(reportId),
+                },
+                {
+                    $set: {
+                        'meta.updatedAt': new Date(),
+                        'meta.updatedBy.by': user,
+                        resolved: newResolvedStatus,
+                    },
+                },
+                {
+                    upsert: false,
+                }
+            )
+    );
+
+    if (modifiedCount === 0)
+        throw createHttpError(
+            400,
+            'Could not update resolved status of feedback report'
+        );
```

Comment:
The modified count would most likely be 0 due to not finding the report, hence a 404.